tests/explainers/test_tree.py

```diff
@@ -1002,8 +752,376 @@ def test_xgboost_buffer_strip():
     assert isinstance(explainer, shap.explainers.Tree)
 
 
+class TestExplainerSklearn:
```

Comment:
Out of interest, what is the motivation behind grouping these tests into a class? Is it purely organisational, or are there other benefits?

Reply:
We have discussed this before. See https://github.com/dsgibbons/shap/pull/88#discussion_r1210869174 for context.

Reply:
PS - one rather minor downside of adding the class-based approach is a complex git history. As with all style changes that cause a big diff, it will be harder to see the evolution of particular tests with `git blame`.

One mitigation could be to add a `.git-blame-ignore-revs` file with a list of commits to ignore. This is another reason to aim for a strict separation style PRs from PRs with functional changes. For more info see:
- python black: [Avoiding ruining git blame](https://black.readthedocs.io/en/stable/guides/introducing_black_to_your_project.html#avoiding-ruining-git-blame)
- Github: [ignore commits in blame view](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)